File: app/src/main/java/com/zawzaw/savethelibrary/utils/TypefaceManager.java
package com.zawzaw.savethelibrary.utils;

import android.content.res.AssetManager;
import android.graphics.Typeface;
import android.util.LruCache;

/**
 * Created by zawzaw on 26/12/17.
 */

public class TypefaceManager
{
    public static final String PYIDAUNGSU = "pyidaungsu.ttf";
    public static final String ZAWGYIONE = "zawgyi_one.ttf";
    public static final String PASSCODEFONT = "Font-Regular.ttf";
    public static final String ROBOTO = "roboto.ttf";
    public static final String DIGIT = "digit.ttf";
    public static final String MYANMAR_SAGAR = "myanmar_sagar.ttf";

    private AssetManager manager;

    private final LruCache<String, Typeface> mCache;

    public TypefaceManager(AssetManager manager)
    {
        this.manager = manager;
        int cacheSize = 4 * 1024 * 1024;
        mCache = new LruCache<>(cacheSize);
    }

    public Typeface getShitUnicode() {
        return getTypeface(PYIDAUNGSU);
    }

    public Typeface getShitZawgyi() {
        return getTypeface(ZAWGYIONE);
    }

    public Typeface getDigitFont() {
        return getTypeface(DIGIT);
    }

    public Typeface getPasscodeFont() {
        return getTypeface(PASSCODEFONT);
    }

    public Typeface getRobotoFont() {
        return getTypeface(ROBOTO);
    }

    public Typeface getMyanmarSager() {
        return getTypeface(MYANMAR_SAGAR);
    }

    public Typeface getTypeface(final String finalname)
    {
        Typeface typeface = mCache.get(finalname);
        if (typeface == null)
        {
            typeface = Typeface.createFromAsset(manager, finalname);
        }
        return typeface;
    }

}

File: app/src/main/java/com/zawzaw/savethelibrary/utils/TypefaceManager.kt
package com.zawzaw.savethelibrary.utils

import android.content.res.AssetManager
import android.graphics.Typeface
import android.util.LruCache

/**
 * Created by zawzaw on 01/01/18.
 */

class TypefaceManager(private val manager: AssetManager) {

    private val mCache: LruCache<String, Typeface>

    val shitUnicode: Typeface
        get() = getTypeface(PYIDAUNGSU)
    val shitZawgyi: Typeface
        get() = getTypeface(ZAWGYIONE)
    val digitFont: Typeface
        get() = getTypeface(DIGIT)
    val passcodeFont: Typeface
        get() = getTypeface(PASSCODEFONT)
    val robotoFont: Typeface
        get() = getTypeface(ROBOTO)
    val myanmarSager: Typeface
        get() = getTypeface(MYANMAR_SAGAR)

    init {
        val cacheSize = 4 * 1024 * 1024
        mCache = LruCache(cacheSize)
    }

    fun getTypeface(filename: String): Typeface {
        var typeface: Typeface? = mCache.get(filename)
        if (typeface == null) {
            typeface = Typeface.createFromAsset(manager, filename)
        }
        return typeface
    }

    companion object {

        val PYIDAUNGSU = "pyidaungsu.ttf"
        val ZAWGYIONE = "zawgyi_one.ttf"
        val PASSCODEFONT = "Font-Regular.ttf"
        val ROBOTO = "roboto.ttf"
        val DIGIT = "digit.ttf"
        val MYANMAR_SAGAR = "myanmar_sagar.ttf"
    }

}
